File: September/13th September,2024/Leetcode(1310).js
/**
 * @param {number[]} arr
 * @param {number[][]} queries
 * @return {number[]}
 */
var xorQueries = function(arr, queries) {
    let temp=[arr[0]]
    let n=arr.length;
    let ans=[]
    for(let i=1;i<n;i++){
        temp.push(temp[i-1]^arr[i])
    }
    for(let i=0;i<queries.length;i++){
        let a=queries[i][0]
        let b=queries[i][1]
        if(a==0) ans.push(temp[b])
        else ans.push(temp[b]^temp[a-1])
    }
    return ans;

};

File: September/13th September,2024/Leetcode(1310).py
class Solution:
    def xorQueries(self, arr: List[int], queries: List[List[int]]) -> List[int]:
        ans=[]
        prexor=[arr[0]]
        n=len(arr)
        for i in range(1,n):
            prexor.append(prexor[i-1]^arr[i])

        for i in queries:
            x=0
            if i[0]==0:
                ans.append(prexor[i[1]])
            else:
                ans.append(prexor[i[1]]^prexor[i[0]-1])
            # ans.append(x)
        # print(prexor)
        return ans
            